### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by khou2020**
Fix all reported compiler warnings in -Wall.

### CODE REVIEW

File: src/cases/e3sm_io_case_F_scorpio.cpp [hunk 1]
```diff
@@ -56,9 +56,9 @@ int e3sm_io_case_F_scorpio::wr_test(e3sm_io_config &cfg,
     return err;
 }
 
-int e3sm_io_case_F_scorpio::rd_test(e3sm_io_config &cfg,
-                                    e3sm_io_decom &decom,
-                                    e3sm_io_driver &driver)
+int e3sm_io_case_F_scorpio::rd_test(e3sm_io_config &cfg E3SM_IO_UNUSED,
```
Comment (wkliao) [inline src/cases/e3sm_io_case_F_scorpio.cpp]: Some files still contain E3SM_IO_UNUSED

Comment (khou2020) [reply src/cases/e3sm_io_case_F_scorpio.cpp]: I can't see it on my side. I think it is just sync delay.

Comment (wkliao) [reply src/cases/e3sm_io_case_F_scorpio.cpp]: A clean PR should not first add E3SM_IO_UNUSED and remove it in a later commit.

Comment (khou2020) [reply src/cases/e3sm_io_case_F_scorpio.cpp]: fixed

File: src/drivers/e3sm_io_driver_h5blob.cpp [hunk 2]
```diff
@@ -605,26 +605,26 @@ int e3sm_io_driver_h5blob::get_vara (int fid,
...
 
-int e3sm_io_driver_h5blob::get_vars (int fid,
-                                   int vid,
-                                   MPI_Datatype itype,
-                                   MPI_Offset *start,
-                                   MPI_Offset *count,
-                                   MPI_Offset *stride,
-                                   void *buf,
-                                   e3sm_io_op_mode mode) {
+int e3sm_io_driver_h5blob::get_vars (int fid ,
```
Comment (wkliao) [inline src/drivers/e3sm_io_driver_h5blob.cpp:608]: Stop using clang format.

Comment (khou2020) [reply src/drivers/e3sm_io_driver_h5blob.cpp]: fixed

**Pull request merged**